tests/unit/test_ws_api.py

```diff
@@ -0,0 +1,88 @@
+import queue
+
+import pytest
+import websockets
+
+from pyfenix import WebsocketsAPI, Event, NoConnectionError
+
+@pytest.fixture
+def event_queue():
+    return queue.Queue()
+
+@pytest.fixture
+def api(event_queue):
+    return WebsocketsAPI(event_queue)
+
+@pytest.fixture
+def server_fac(unused_tcp_port):
+    def make_server(impl):
+        return (
+                   websockets.serve(impl, "127.0.0.1", unused_tcp_port),
+                   unused_tcp_port
+               )
+    return make_server
+
+async def test_when_server_is_not_running_will_fail_connect(unused_tcp_port):
+    event_queue = queue.Queue()
+    api = WebsocketsAPI(event_queue)
+    await api.connect(("localhost", unused_tcp_port))
+    assert event_queue.get_nowait() == (Event.CONN_FAIL, "")
+
+async def test_when_recv_will_queue_event(api, event_queue, server_fac):
+    async def send_yay(conn):
+        await conn.send('{"type": "msg_send", "message": "yay"}')
+
+    server, port = server_fac(send_yay)
+    async with server:
+        await api.connect(("localhost", port))
+        await api.recv_event()
+        await api.close()
+
+    assert event_queue.get_nowait() == (Event.MSG_RECV, "yay")
+
+async def test_ignores_empty_messages(api, event_queue, server_fac):
+    async def send_nothing(conn):
+        await conn.send('{"type": "msg_send", "message": ""}')
+
+    server, port = server_fac(send_nothing)
```

Comment:
`send_nothing` is a function, which is passed into `server_fac`, whose argument is `unused_tcp_port`.  Something is named incorrectly.

Reply:
This is pytest fixture hax. Pylint throws a fit about it but I haven't looked up whether you can name a pytest fixture differently than the implementation function.